lightningd/options.c

```diff
@@ -226,24 +237,6 @@ static char *opt_add_announce_addr(const char *arg, struct lightningd *ld)
 		return tal_fmt(NULL, "address '%s' is not announcable",
 			       arg);
 
-	/* gossipd will refuse to announce the second one, sure, but it's
-	 * better to check and fail now if they've explicitly asked for it. */
-	wn = &ld->proposed_wireaddr[n].u.wireaddr;
-	for (size_t i = 0; i < n; i++) {
-		const struct wireaddr *wi;
-
-		if (ld->proposed_listen_announce[i] != ADDR_ANNOUNCE)
-			continue;
-		assert(ld->proposed_wireaddr[i].itype == ADDR_INTERNAL_WIREADDR);
-		wi = &ld->proposed_wireaddr[i].u.wireaddr;
-
-		if (wn->type != wi->type)
-			continue;
-		return tal_fmt(NULL, "Cannot announce address %s;"
-			       " already have %s which is the same type",
-			       type_to_string(tmpctx, struct wireaddr, wn),
-			       type_to_string(tmpctx, struct wireaddr, wi));
-	}
```

Comment:
I'm missing why we are removing this one?

Reply:
Because it is just preventing the user from doing something he must have done consciously. If the user gives multiple addresses he most likely has a reason for it and since it is allowed from the protocol spec, why prevent that?

Reply:
Yes, it used to be forbidden, but spec was updated to allow it (and other implementations allowed it).

Reply:
Ah, got it :) thanks and sorry for the missing